pkg/operator/ceph/cluster/controller.go

```diff
@@ -366,6 +366,8 @@ func (c *ClusterController) configureLocalCephCluster(namespace, name string, cl
 			if err != nil {
 				failedMessage = fmt.Sprintf("failed to create cluster in namespace %q. %v", cluster.Namespace, err)
 				logger.Errorf(failedMessage)
+				// reset c.devicesInUse to avoid issues when updating the CR to fix the cluster
+				c.devicesInUse = false
```

Comment:
I'd propose we just remove the `devicesInUse` check. When the OSDs are being provisioned, if one cluster has already provisioned the OSDs, the new cluster will skip them since it will see they are already in use. So I don't think we need this check anymore. Coordinating the flag across multiple cluster instances is more complicated than it's worth. Someone also might want `useAllDevices: true` in multiple clusters and they would keep them on separate nodes by using the node labels and selectors.